docs/core/extensions/generic-host.md

```diff
@@ -0,0 +1,143 @@
+---
+title: .NET Generic Host
+author: IEvangelist
+description: Learn about the .NET Generic Host, which is responsible for app startup and lifetime management.
+ms.author: dapine
+ms.date: 09/18/2020
+---
+
+# .NET Generic Host
+
+The Worker Service templates create a .NET Generic Host, <xref:Microsoft.Extensions.Hosting.HostBuilder>. The Generic Host can be used with other types of .NET applications, such as Console apps. A Generic Host is not specific to Web applications, for Web apps use the <xref:Microsoft.AspNetCore.Hosting.WebHostBuilder?displayProperty=nameWithType>. For more information on the web host, see [ASP.NET Core Web Host](/aspnet/core/fundamentals/host/web-host).
+
+## Host definition
+
+A *host* is an object that encapsulates an app's resources, such as:
+
+- Dependency injection (DI)
+- Logging
+- Configuration
+- `IHostedService` implementations
+
+When a host starts, it calls <xref:Microsoft.Extensions.Hosting.IHostedService.StartAsync%2A?displayProperty=nameWithType> on each implementation of <xref:Microsoft.Extensions.Hosting.IHostedService> registered in the service container's collection of hosted services. In a worker service app, all `IHostedService` implementations that contain <xref:Microsoft.Extensions.Hosting.BackgroundService> instances have their <xref:Microsoft.Extensions.Hosting.BackgroundService.ExecuteAsync%2A?displayProperty=nameWithType> methods called.
+
+The main reason for including all of the app's interdependent resources in one object is lifetime management: control over app startup and graceful shutdown.
```

Comment:
It might be good to mention the NuGet package somewhere in the intro/overview also.